### PR TITLE
[RSDK-7179]: Module termination only responds to SIGKILL when device is unplugged

### PR DESCRIPTION
**Pull request by hexbabe**
https://viam.atlassian.net/browse/RSDK-7179

Bug details here ^

tl;dr is the module doesn't clean up properly and we are relying on SIGKILL to exit

In this PR, I also bump Go and Go deps in integration tests and do general code structure cleanup for maintainability. Most of the code additions are in go.sum. Sorry it is a bit messy-- but all good changes I hope!

### CODE REVIEW

File: src/worker.py [hunk 1]
```diff
@@ -180,19 +146,31 @@ def __init__(
         self.reconfigure = reconfigure
         self.logger = logger
 
-        self.color_image = None
```
Comment (hexbabe) [inline src/worker.py:183]: This stuff is totally deprecated. I should've removed ages ago

File: src/oak.py [hunk 2]
```diff
@@ -266,9 +267,9 @@ def reconfigure(
         """
         cls: Oak = type(self)
         try:
-            LOGGER.debug("Trying to stop worker.")
-            cls.worker.stop()
-            LOGGER.info("Stopping active worker and reconfiguring...")
+            LOGGER.debug("Trying to stop worker manager and worker.")
+            cls.worker_manager.stop()
```
Comment (hexbabe) [inline src/oak.py:271]: Lifecycle of `worker` now completely managed by worker manager.

File: src/oak.py [hunk 3]
```diff
@@ -540,6 +544,26 @@ async def get_properties(
...
+
+        """
+        cls: Oak = type(self)
+        attempts = 0
+        while attempts < max_attempts:
+            if cls.worker.running:
+                return
+            time.sleep(sleep_time)
+            attempts += 1
+        raise ViamError("camera data requested before camera worker was ready.")
```
Comment (randhid) [inline src/oak.py:565]: add more user facing instructions to the error, what do i do here?

**Review by randhid — APPROVED** [review]
Really like this. Can we add a test to test the bad case atl all? To see that we're gracefully handling a Reconfigure condition?

Can you also, for my peace of mind, ask for a server restart while it's reconfiguring? Just to see if it's hanging.

File: integration-tests/go.mod [hunk 4]
```diff
@@ -1,159 +1,174 @@
 module github.com/viamrobotics/viam-camera-realsense/integration
 
-go 1.19
+go 1.21
+
+toolchain go1.22.1
```
[the remark this answers is not in the document]
Comment (hexbabe) [reply integration-tests/go.mod:5]: I am not sure... I ran `go get -u ./...` and this new diff was generated that way

File: src/worker_manager.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+from typing import Callable
+from logging import Logger
+
+from src.worker import Worker
+
+
+class WorkerManager(Thread):
+    """
+    WorkerManager watches and manages the lifetime of the Worker to make sure
+    the connection to the camera is healthy, as well as terminating it when necessary.
```
Comment (hexbabe) [inline src/worker_manager.py:12]: This comment is outdated. Reminder for myself to update it

File: src/oak.py (hunk 3 above)
Comment (hexbabe) [reply src/oak.py]: I think the new message suffices. LMK ur thoughts

**Comment by hexbabe** [comment]
> Really like this. Can we add a test to test the bad case atl all? To see that we're gracefully handling a Reconfigure condition?
> 
> Can you also, for my peace of mind, ask for a server restart while it's reconfiguring? Just to see if it's hanging.

It's difficult to test the disconnected camera logic because I have no idea/if it's even possible to simulate a camera disconnect. Here is a [video](https://www.loom.com/share/b6c3ee6fdcc34eeba7542e0bb37b788b?sid=0257a7bc-c4d6-42b1-a5c3-7d790b837585)

File: integration-tests/go.mod (hunk 4 above)
[the remark this answers is not in the document]
Comment (randhid) [reply integration-tests/go.mod:5]: let's keep an eye on it for the different system builds, what was this built on?
here's what toolchains do btw - introduced in go 1.21 
https://go.dev/doc/toolchain

[the remark this answers is not in the document]
Comment (hexbabe) [reply integration-tests/go.mod:5]: my mac. taking a look

[the remark this answers is not in the document]
Comment (hexbabe) [reply integration-tests/go.mod:5]: I'm gonna take it out. There's no particular reason why we need to use the compiler of a different version of Go in our case

**Comment by hexbabe** [comment]
Are you good with me merging this now @randhid ?

**Comment by randhid** [comment]
> Are you good with me merging this now @randhid ?
> 
> I'm gonna make a new release after this is on main since this is a pretty big fix

sorry to leave you hanging, yes video was excellent

**Pull request merged**